### ISSUE TITLE
Documentation: form accessibility reference to closed django issues

### ISSUE THREAD

**Issue by Chadys** [wagtail/wagtail#12009]
### Pertinent section of the Wagtail docs

https://docs.wagtail.org/en/stable/advanced_topics/accessibility_considerations.html#forms

### Details

In the documentation about forms accessibility, some considerations are raised that directly references django tickets. Some of those tickets have since been closed, so perhaps the considerations are outdated and should be removed? If that is not the case, then those django tickets were not the sole culprit for these issues and we may want to make it more clear in the docs.

Affected considerations are: 
- "Avoid rendering helpers such as as_table, as_ul, as_p" (Django ticket [#32339](https://code.djangoproject.com/ticket/32339) )
- "group related fields together in fieldset" (Django ticket [#32338](https://code.djangoproject.com/ticket/32338) )

### Working on this

Anyone can contribute to this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

**Comment by lb-** [comment]
One thing to consider is that Wagtail may be running supported versions of Django where those newer fixes/features are not yet able to be leveraged.

**Comment by Chadys** [comment]
Right! Then, if a supported version is affected (I will check) maybe simply going from

> (see Django ticket #X)

to

> (for Django < X.X.X, see Django ticket #X)

**Comment by Chadys** [comment]
Wagtail currently support Django 4.2.x and 5.0.x

- "Avoid rendering helpers such as as_table, as_ul, as_p" (Django ticket [#32339](https://code.djangoproject.com/ticket/32339) ) -> Fix is in 5.0 so the note about the version is pertinent
- "group related fields together in fieldset" (Django ticket [#32338](https://code.djangoproject.com/ticket/32338) ) -> Fix is in 4.1 so Wagtail is no longer impacted by this in supported Django version

**Comment by gasman** [comment]
As far as I read [#32339](https://code.djangoproject.com/ticket/32339), the eventual resolution was to introduce a new `as_div` format following better accessibility practices, and demote the coverage of `as_table`/`as_ul`/`as_p` in the docs (possibly along with switching the default rendering to `as_div`? This isn't clear to me).

The `as_table`/`as_ul`/`as_p` helpers still exist, and are as problematic as they ever were (since there is no way for Django to fix them without breaking a huge amount of existing Django project code), so it seems to me that the advice to avoid them is still valid, and the link to the closed issue is still a relevant source of information for anyone wanting to know the reasoning for that advice. At most, the advice might be less _necessary_ now that Django is promoting better alternatives.

The situation with [#32338](https://code.djangoproject.com/ticket/32338) is less clear to me - it sounds like Django now adds a `<fieldset>` around radio buttons and checkboxes unless the legacy `as_table`/`as_ul`/`as_p` helpers are being used, so while the general advice to use fieldset for grouping related elements is still valid, the specific mention of radios and checkboxes might be redundant now.

In short - I believe the links to the Django issues were meant as additional context for the advice rather than singling them out as the "culprit" of the accessibility failures, and the closed status of the issues is not itself an indication that the advice is no longer relevant. (But having said that, there might well be better references for "here's why this is best practice" than a link about how Django used to get it wrong...)

I think it will ultimately be up to @thibaudcolas to decide what changes (if any) need to be made to this advice in the light of the Django fixes, but anyone interested in this could help by researching:

* What specific changes were made to Django's behaviour in these fixes
* How these affect the Wagtail form builder in particular
* Whether these changes mean that it's unlikely or impossible for the mentioned accessibility failures to happen, and thus it's no longer necessarily to warn against them.

**Comment by kamalakiruthi8** [comment]
Hi! I’m currently exploring ways to contribute to Wagtail and would be happy to help with this issue.

I can look into the referenced Django tickets (#32339 and #32338), review the changes introduced there, and check how they affect Wagtail’s form builder and the related accessibility guidance in the documentation.

If that sounds helpful, I’d be glad to investigate and propose an update to the docs.
